File: trunk/theLink/example/ruby/testserver.rb
#+
#§  \file       theLink/example/ruby/testserver.rb
#§  \brief      \$Id: LbMain 26 2009-12-03 11:48:43Z aotto1968 $
#§  
#§  (C) 2010 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev: 26 $
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§
require "rubymsgque"
class testserver(MqS):
  def GTCX(ctx):
    ctx.SendSTART()
    ctx.SendI(ctx.LinkGetCtxId())
    ctx.SendC("+")
    if (ctx.LinkIsParent()):
      ctx.SendI(-1)
    else:
      ctx.SendI(ctx.LinkGetParent().LinkGetCtxId())
    ctx.SendC("+");
    ctx.SendC(ctx.ConfigGetName());
    ctx.SendC(":")
    ctx.SendRETURN()
  def ServerConfig(ctx):
    ctx.ServiceCreate("GTCX",ctx.GTCX)

if __name__ == "__main__":
  srv = testserver()
  try:
    srv.ConfigSetName("testserver")
    srv.ConfigSetServerSetup(srv.ServerConfig)
    srv.ConfigSetFactory(lambda: testserver())
    srv.LinkCreate(sys.argv)
    srv.ProcessEvent(wait="FOREVER")
  except:
    srv.ErrorSet()
  finally:
    srv.Exit()

File: trunk/theLink/example/java/testserver.java
/**
 *  \file       theLink/example/java/testserver.java
 *  \brief      \$Id$
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev$
 *  \author     EMail: aotto1968 at users.berlios.de
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */
package example;

import javamsgque.*;

final class testserver extends MqS implements IServerSetup, IFactory {
  public MqS Factory() {
    return new testserver();
  }

  class GTCX implements IService {
    public void Service (MqS ctx) throws MqSException {
      SendSTART();
      SendI(LinkGetCtxId());
      SendC("+");
      if (LinkIsParent()) {
	SendI(-1);
      } else {
	SendI(LinkGetParent().LinkGetCtxId());
      }
      SendC("+");
      SendC(ConfigGetName());
      SendC(":");
      SendRETURN();
    }
  }

  public void ServerSetup() throws MqSException {
    ServiceCreate("GTCX", new GTCX());
  }

  public static void main(String[] argv) {
    MqS.Init("java", "example.testserver");
    testserver srv = new testserver();
    try {
      srv.LinkCreate(argv);
      srv.ProcessEvent(MqS.WAIT.FOREVER);
    } catch (Throwable e) {
      srv.ErrorSet(e);
    } finally {
      srv.Exit();
    }
  }
}



